### ISSUE TITLE
In place editing corrupts file

### ISSUE THREAD

**Issue by dunha** [gpac/gpac#1924]
MP4Box 1.1.0-DEV compiled standalone from a recent download of master branch
Updating handler names for subtitle tracks
mp4box -name 3=SubtitleHandler -name 5=NewHandlerName input.mp4

The process works and data seems to be OK, the video plays, but there seems to be some corruption, and no further processing is possible
 mp4box -info input.mp4
[iso file] Unknown top-level box type 9.2.
[iso file] Incomplete box 9.2. - start 577023550 size 1635149354
[iso file] Incomplete file while reading for dump - aborting parsing

The rest of the output is as expected

A previous attempt editing only one handler name referenced box type .100

ffmpeg doesn't appear bothered

**Comment by jeanlf** [comment]
Could you upload your test file? I cannot reproduce here

**Comment by dunha** [comment]
I've uploaded the original file plus a copy (suffix 1) after the process

command
 mp4box -name 3=SubtitleHandler -name 5=VideoTripper -lang 5=eng "D:\temp\M074_1-0 slk1.mp4"
Saving D:\temp\M074_1-0 slk1.mp4: In-place rewrite

[WeTransfer](https://we.tl/t-WtbqhcJ0st)

**Comment by jeanlf** [comment]
indeed, fixed by d045df83
Thanks for the report and the file.

**Comment by dunha** [comment]
Sorry, not fixed for me 
MP4Box - GPAC version 1.1.0-DEV-rev1216-gb39aa09c0-master

**Comment by jeanlf** [comment]
you're not on the right version that commit is much older than the patch, you need at least rev1309

**Comment by dunha** [comment]
Yes, sorry it wasn't updated

λ mp4box -version
MP4Box - GPAC version 1.1.0-DEV-rev1314-gb9b5abb0e-master

now the operation doesn't seem to finish and my file grows (22 Gb when I cancelled) hugely. It was in OneDrive so I thought that was a problem during syncing. I recovered the earlier version but eventually lost it on Onedrive. I had copied a 22 Gb version which Mp4box would not read it but ffmpeg recovered it, so I tried it from the regular file system but same problem
